Sources/Controllers/TargetMenu/GPX/TrackMenu/OATrackMenuHudViewController.mm

```diff
@@ -149,10 +180,24 @@ - (void)viewDidLoad
     [self startLocationServices];
 }
 
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
+{
+    [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+    [coordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext>  _Nonnull context) {
+
+    } completion:^(id<UIViewControllerTransitionCoordinatorContext>  _Nonnull context) {
+        _routeLineChartHelper.isLandscape = [self isLandscape];
+        _routeLineChartHelper.screenBBox = CGRectMake(0., 0.,
+                DeviceScreenWidth - ([self isLandscape] ? DeviceScreenWidth * 0.45 : 0.0),
+                DeviceScreenHeight - ([self isLandscape] ? 0.0 : [self getViewHeight]));
+    }];
+}
+
 - (void)hide:(BOOL)animated duration:(NSTimeInterval)duration onComplete:(void (^)(void))onComplete
 {
     [super hide:YES duration:duration onComplete:^{
         [self stopLocationServices];
+        [self.mapViewController.mapLayers.routeMapLayer hideCurrentStatisticsLocation];
```

Comment:
Why do you use routeMapLayer for this? The gpx tracks are handled in the OAGPXLayer